bc_obps/registration/middleware/current_user_middleware.py

```diff
@@ -13,7 +14,7 @@ def __call__(self, request):
         auth_header = request.headers.get('Authorization')
         if auth_header:
             try:
-                user_guid = json.loads(auth_header).get('user_guid')
+                user_guid = UUID(json.loads(auth_header).get('user_guid'))
```

Comment:
New! From Sepehr: We are receiving user_guid from the front-end as a UUID without hyphens. We store it as the cache key. However, when we update the user, we attempt to invalidate their cache using the user_guid with hyphens, causing a mismatch and preventing successful invalidation.